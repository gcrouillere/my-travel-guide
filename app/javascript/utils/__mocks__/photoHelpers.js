let photo = {
  bytes: 1000,
  created_at: "2018-08-11T12:24:32Z",
  etag: "abcdefghijklmnopqrstuvwxyz12345",
  format: "jpg",
  height: 1000,
  public_id: "123",
  resource_type: "image",
  signature: "abcdefghijklmnopqrstuvwxyz12345",
  tags: ["mytravelguide"],
  type: "upload",
  url: "http://res.cloudinary.com/demo/image/upload/v1473596672/eneivicys42bq5f2jpn2.jpg",
  version: 123,
  width: 1000
}

const photoUpload = jest.fn(() => Promise.resolve(photo));

module.exports = {
  photoUpload,
  photo
}
