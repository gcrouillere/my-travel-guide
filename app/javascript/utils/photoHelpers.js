import $ from 'jquery'
import { CLOUDINARYKEYS } from './../config/config'

export default {

  photoUpload(event, formData, context) {
    var self = context
    const data = $.ajax({
    xhr: function() {
        var xhr = new window.XMLHttpRequest()
        //Upload progress
        xhr.upload.addEventListener("progress", function(evt){
          if (evt.lengthComputable) {
            var percentComplete = evt.loaded / evt.total * 100;
            self.setState({progress: percentComplete})
          }
        }, false)
        return xhr;
      },
      method: 'POST',
      url: `https://api.cloudinary.com/v1_1/${CLOUDINARYKEYS.cloudName}/upload`,
      data: formData,
      processData: false,
      contentType: false,
      succes: function(data) { return data },
      error: function(xhr, error) { console.log(error) }
    })
    return data
  }

}
