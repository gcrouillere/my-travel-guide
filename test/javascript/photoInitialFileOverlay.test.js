import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router'
import renderer from 'react-test-renderer';

import PhotoInitialFileOverlay from '../../app/javascript/packs/articleForm/contentMenu/photoInitialFileOverlay'
import photoHelpers from '../../app/javascript/utils/photoHelpers'

jest.mock('../../app/javascript/utils/photoHelpers')

configure({ adapter: new Adapter() });

describe('PhotoInitialFileOverlay test suite', () => {

  let abandonPhotoCreation, printLocation, addNewPhotoBloc
  abandonPhotoCreation = printLocation = addNewPhotoBloc = jest.fn()

  const photoInitialFileOverlayTree = renderer
    .create(<PhotoInitialFileOverlay abandonPhotoCreation={abandonPhotoCreation}
      printLocation={printLocation} addNewPhotoBloc={addNewPhotoBloc} />)
    .toJSON()
  expect(photoInitialFileOverlayTree).toMatchSnapshot();

  const wrapper = mount(<PhotoInitialFileOverlay abandonPhotoCreation={abandonPhotoCreation} printLocation={printLocation}
    addNewPhotoBloc={addNewPhotoBloc} />)

  it('loads a photo on input change', async () => {
    const photo = photoHelpers.photo
    await wrapper.instance().onPhotoSelected({ target: { files: [{name: "my photo"}] }})

    expect(addNewPhotoBloc).toHaveBeenCalledWith(photo)
    expect(abandonPhotoCreation).toHaveBeenCalled()
    expect(wrapper.state('progress')).toEqual(0)
    expect(wrapper.state('fileName')).toEqual(null)
  })

  it('builds a formData on function call', () => {
    const formData = wrapper.instance().buildFormData({ target: { files: [{name: "my photo"}] }})
    expect(formData).toEqual({
      file: { name: 'my photo' },
      upload_preset: expect.anything(),
      tags: 'mytravelguide,my photo',
      context: 'photo=my photo'
    })
    expect(wrapper.state('fileName')).toEqual('my photo')
    expect(wrapper.find(".custom-file-label").text()).toEqual('my photo')
  })

  it('passes overlay closing to upper component', () => {
    wrapper.find(".close").simulate('click')
    expect(abandonPhotoCreation).toHaveBeenCalled()
  })

})
