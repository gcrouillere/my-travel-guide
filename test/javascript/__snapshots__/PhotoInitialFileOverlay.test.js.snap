// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="photoInitialFileOverlay "
>
  <button
    aria-label="Close"
    className="close"
    onClick={[Function]}
  >
    <span
      aria-hidden="true"
    >
      Ã—
    </span>
  </button>
  <p
    className="mapOverlayTitle"
  >
    Select a photo on your disk
  </p>
  <div
    className="input-group mb-3"
  >
    <div
      className="custom-file"
    >
      <input
        accept="image/*"
        aria-describedby="inputGroupFileAddon01"
        className="custom-file-input"
        id="fileupload"
        multiple={false}
        onChange={[Function]}
        type="file"
      />
      <label
        className="custom-file-label"
        htmlFor="fileupload"
      >
        Click here to browse your files
      </label>
    </div>
  </div>
</div>
`;
